### DISCUSSION TITLE
M5StickC+ Client only 1/4 display

### DISCUSSION THREAD

**Discussion by jumperger** [josephdadams/TallyArbiter#386] [Q&A] [answered]
Hi, I keep having Problems to run my M5StickC+ Client.
Using the Arduino IDE to flash the M5 did not work for me I couldn't connect to the WiFi.
Using the esphome software to flash the M5 worked, but I only get 1/4 of the screen , please see attched picture.

 I got the AP and could do the Setup using my phone
 I can see the M5StickC+ in the TallyArbiter webinterface , so it has got an IP from the router and connected to the TA-server.
 In producer view I may activate the Flash button and the M5StickC+ flashes!    
BUT the display looks like this, and it's the white upper right rectangle which is flashing when I hit Flash
    M5StickC+_connected
![M5StickC+_connected](https://user-images.githubusercontent.com/55156091/155430964-ce3adbb6-d2d1-483b-9189-189ef1809406.jpg)
When changing Input1 status:
-    deselecting the Input 1 on my ATEM, the white rectangle turns BLACK
-    Input 1 in Preview (green) on my ATEM, the white rectangle turns Purple-White
-    Input 1 in Program (red) on my ATEM, the white rectangle turns WHITE

I can't take a picture from the color difference PrupleWhite vs WHITE

**long PRESS on M5** button brings the following screen:
![M5StickC+_M5button](https://user-images.githubusercontent.com/55156091/155432211-81a796c0-d97b-4fef-8899-c9412f353a7d.jpg)
which is the end of the mirrored IP-address of the M5StickC+

Is this a known issue or did I do something the wrong way?

**Comment by josephdadams** [comment]
@MatteoGheza any ideas? Not the first time I’ve seen someone report this.

**Reply by jumperger** [reply]
Could I do some testing for you?
I have 4 M5StickC+
Info: 
the M5StickC+ was flashed with M5Stack firmware 1.9.2 before flashing the TA-Client firmware with esphome.exe

**Reply by jumperger** [reply]
@josephdadams 
dear Joseph, could you please take a look st this post : 

https://community.home-assistant.io/t/m5stack-core2-m5stick-cplus-esphome-support/387554/4

They are discussing problems with the C+ and  Core2 display, with esphome. I’m a real newbie and do not understand what they say. You probably do understand it. 
Perhaps this may help investigating.

**Reply by jumperger** [reply]
@josephdadams @MatteoGheza 
If I can be of any help, for testing or further informations, please let me know.
regards
jumperger

**Reply by commputethis** [reply]
The same thing happens for me with M5StickC+ when using esphome to flash the device.  If I use Arduino IDE, I do not have this issue.

**Reply by commputethis** [reply, marked as answer]
I found what is causing this, but not sure how to fix.  I found it when using an M5StickC Plus with a different program.  The process to auto-create the files on Github to use with esphome software is using M5StickC.h and not M5StickCPlus.h.  It is why we need to set the first line of m5stickc-listener.ino file to a 1 when using the M5StickC Plus.  The automated process does not take the different M5StickCs into account.  It is only doing it for the non "plus" version. 

I'm not sure if I used the correct terminology here, but hope it makes sense.

**Comment by Jocke4f** [comment]
The challenge is that the same code base (.ino file) supports all the three existing variants of M5StickC so the proper preprocessor flags needs to be defined. This is handled in Arduino IDE by selecting the proper board type.
